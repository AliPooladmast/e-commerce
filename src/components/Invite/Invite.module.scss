@use "src/styles";

.Container {
  padding: 30px 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background-color: lightgrey;
  box-shadow: styles.$shadow;

  h1 {
    font-size: 40px;
    margin-bottom: 20px;

    @media only screen and (max-width: 600px) {
      font-size: 30px;
    }
  }

  button {
    padding: 10px 20px;
    border: none;
    border-radius: 5px;
    background-color: teal;
    color: white;
    cursor: pointer;
    font-size: 20px;
    box-shadow: styles.$shadow;

    @media only screen and (max-width: 600px) {
      font-size: 16px;
    }

    &:hover {
      background-color: rgb(0, 156, 156);
    }
  }
}

.Desc {
  font-size: 24px;
  margin-bottom: 30px;
  font-weight: 300;
  padding: 0 20px;

  @media only screen and (max-width: 600px) {
    text-align: center;
    font-size: 20px;
  }
}
