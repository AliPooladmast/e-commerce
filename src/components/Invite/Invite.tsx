import React from "react";
import style from "./Invite.module.scss";
import { Link } from "react-router-dom";

const Invite = () => {
  return (
    <div className={style.Container}>
      <h1>Join Now!</h1>
      <div className={style.Desc}>
        by registering here you can start buying products today!
      </div>
      <Link data-testid="register-link" to="/register">
        <button>Register</button>
      </Link>
    </div>
  );
};

export default Invite;
