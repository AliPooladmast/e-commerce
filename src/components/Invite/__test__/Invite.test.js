import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import Invite from "../Invite";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

it("snapshot of component", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Invite />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("check if the link element contains register path", async () => {
  render(
    <BrowserRouter>
      <Invite />
    </BrowserRouter>
  );

  expect(screen.getByTestId("register-link").href).toMatch(/register/i);
});
