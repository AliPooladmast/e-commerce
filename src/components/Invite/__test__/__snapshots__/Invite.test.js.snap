// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component 1`] = `
<div
  className="Container"
>
  <h1>
    Join Now!
  </h1>
  <div
    className="Desc"
  >
    by registering here you can start buying products today!
  </div>
  <a
    data-testid="register-link"
    href="/register"
    onClick={[Function]}
  >
    <button>
      Register
    </button>
  </a>
</div>
`;
